erpc/networks.go

```diff
@@ -227,41 +228,27 @@ func (n *Network) EvmLeaderUpstream(ctx context.Context) common.Upstream {
 func (n *Network) getFailsafeExecutor(req *common.NormalizedRequest) *FailsafeExecutor {
 	method, _ := req.Method()
 	finality := req.Finality(context.Background())
+	networkId := n.networkId
 
-	// First, try to find a specific match for both method and finality
-	for _, fe := range n.failsafeExecutors {
-		if fe.method != "*" && len(fe.finalities) > 0 {
-			matched, _ := common.WildcardMatch(fe.method, method)
-			if matched && slices.Contains(fe.finalities, finality) {
-				return fe
-			}
-		}
-	}
-
-	// Then, try to find a match for method only (empty finalities means any finality)
-	for _, fe := range n.failsafeExecutors {
-		if fe.method != "*" && len(fe.finalities) == 0 {
-			matched, _ := common.WildcardMatch(fe.method, method)
-			if matched {
-				return fe
-			}
-		}
+	// Get params from JsonRpcRequest
+	var params []interface{}
+	if jrpcReq, err := req.JsonRpcRequest(); err == nil && jrpcReq != nil {
+		params = jrpcReq.Params
 	}
 
-	// Then, try to find a match for finality only
 	for _, fe := range n.failsafeExecutors {
-		if fe.method == "*" && len(fe.finalities) > 0 {
-			if slices.Contains(fe.finalities, finality) {
+		if fe.config != nil && len(fe.config.Matchers) > 0 {
+			matcher := matchers.NewConfigMatcher(fe.config.Matchers)
+			result := matcher.MatchRequest(networkId, method, params, finality)
+			if result.Matched && result.Action == common.MatcherInclude {
```

Comment:
For evaluation probably we can have a simpler external interface?
```
if matcher.Match(fe.config.Matchers, req) {
  // ...
}
```

Then we can avoid the req handling above as well?